### PR TITLE
Feature/calypso reader comments

### LINKED ISSUE #3060: Reader comment redesign
Details to come

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #3060 - redesigns reader comments to more closely match the Calypso web reader.

calypso-comments

### CODE REVIEW

**Comment by tonyr59h** [comment]
After clicking Reply to reply to a comment I notice the input EditText changes to `Reply to comment...` but I can't figure out how to cancel that and change back to `Reply to post...`

**Comment by tonyr59h** [comment]
I'm not sure if these are problems with the Reader or Comments:

* [This image](https://cldup.com/4myvyHPbT5.png) shows an error with the comment tree. I posted a bunch of comments, incrementing the number after `Test` each time. Somehow `Test4/5` ended up nested under a later comment, `Test7`.
* I deleted all those `Test*` comments then went back to add more to the article. All the deleted comments were still there.

**Comment by nbradbury** [comment]
> After clicking Reply to reply to a comment I notice the input EditText changes ... 
> I posted a bunch of comments, incrementing the number after Test each time. Somehow Test4/5 ended up nested under a later comment...

This PR doesn't touch those areas, which concerned me. Sure enough, it turns out both of those bugs existed prior to this PR :( Can you file those as separate issues?

> I deleted all those Test* comments then went back to add more to the article.

Where did you delete the comments from?

**Comment by tonyr59h** [comment]
> Can you file those as separate issues?

You bet!

> Where did you delete the comments from?

Just went to the Comments page from the My Sites fragment and trashed them all individually.

**Comment by nbradbury** [comment]
> Just went to the Comments page from the My Sites fragment and trashed them all individually.

Well :hankey:. I just reproduced that - in develop. So another bug that existed prior to this redesign ;(

**Comment by tonyr59h** [comment]
> So another bug that existed prior to this redesign ;(

New issue incoming! Well if all my issues are already in develop I'm happy to close. Just about finished with the code review right now.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderCommentsPostHeaderView.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        initView(context);
+    }
+
+    public ReaderCommentsPostHeaderView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initView(context);
+    }
+
+    private void initView(Context context) {
+        View view = inflate(context, R.layout.reader_comments_post_header_view, this);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderCommentsPostHeaderView.java]: `view` isn't being used here.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderCommentsPostHeaderView.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public void setPost(final ReaderPost post) {
+        if (post == null) return;
+
+        TextView txtTitle = (TextView) findViewById(R.id.text_post_title);
+        TextView txtBlogName = (TextView) findViewById(R.id.text_blog_name);
+        TextView txtDateline = (TextView) findViewById(R.id.text_post_dateline);
+        TextView txtCommentsClosed = (TextView) findViewById(R.id.text_comments_closed);
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderCommentsPostHeaderView.java]: `txtCommentsClosed` isn't being used.

File: WordPress/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -800,8 +800,8 @@
     <string name="reader_btn_unfollow">Following</string>
 
     <!-- EditText hints -->
-    <string name="reader_hint_comment_on_post">Add your comment</string>
-    <string name="reader_hint_comment_on_comment">Reply to comment</string>
+    <string name="reader_hint_comment_on_post">Reply to post...</string>
+    <string name="reader_hint_comment_on_comment">Reply to comment...</string>
```
Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: The three dots here should probable be switched to the ellipsis character, `…` (or \&#8230;).

**Comment by nbradbury** [comment]
> view isn't being used here.
> txtCommentsClosed isn't being used.
> The three dots here should probable be switched to the ellipsis character

All corrected in 27e6743.

**Comment by tonyr59h** [comment]
:shipit:

**Comment by tonyr59h** [comment]
Merging into 4.5 as well.

**Pull request merged**